stan/math/prim/mat/fun/cholesky_decompose.hpp

```diff
@@ -25,10 +29,18 @@ Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic> cholesky_decompose(
     const Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>& m) {
   check_square("cholesky_decompose", "m", m);
   check_symmetric("cholesky_decompose", "m", m);
+#ifdef STAN_OPENCL
```

Comment:
We might want to put the size check here, but probably only if there isn't a `cl::Buffer` on the Eigen matrix.

Reply:
In case the data is already on the GPU, we dont want to copy the data back to the CPU and calculate on the CPU. We would just copy the result back (once we figure out shared virtual memory not even that). I believe that is the reasoning here. I am just guessing tho.

Reply:
I'm not sure which order you planned to merge these things. If we merge this first, agreed we can't check for the cl::Buffer, if we merge this second, we should add the cl::Buffer check to the other PR.

Reply:
Let's merge this first. There's a couple other little things in the cache stuff I want to work on before we merge that PR (I'll write about that in the cache PR later today or tmrw)